core/iwasm/include/wasm_export.h

```diff
@@ -84,6 +84,9 @@ typedef struct wasm_export_type {
     wasm_import_export_kind_t kind;
 } wasm_export_type;
 
+struct WASMFuncType;
+typedef struct WASMFuncType *wasm_function_t;
```

Comment:
The related type has been defined in core/iwasm/include/gc_export.h:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/120b9659dd4f3f500419e05fceab1f47aba54529/core/iwasm/include/gc_export.h#L138
And some similar APIs are also declared in gc_export.h and implemented in core/iwasm/common/gc/gc_common.c when GC feature is enabled:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/include/gc_export.h#L240-L277

How about re-using the definition wasm_func_type_t:
```C
struct WASMFuncType;
typedef struct WASMFuncType *wasm_func_type_t;
```
And define the APIs:
```C
WASM_RUNTIME_API_EXTERN wasm_func_type_t
wasm_runtime_get_export_func_type(wasm_module_t const module, const char *name);

WASM_RUNTIME_API_EXTERN wasm_func_type_t
wasm_runtime_get_import_func_type(wasm_module_t const module, const char *name);

/* Get the param type array and the param type count */
WASM_RUNTIME_API_EXTERN wasm_valkind_t *
wasm_func_type_get_param_types(const wasm_func_type_t func_type, uint32 *p_param_count);

/* Get the result type array and the result type count */
WASM_RUNTIME_API_EXTERN wasm_valkind_t *
wasm_func_type_get_result_types(const wasm_func_type_t func_type, uint32 *p_result_count);
```